### PR TITLE
Secrets and context available within the components

### LINKED ISSUE #9: Make secrets available in event.secrets

### LINKED ISSUE #10: Make context available in event.context

### PR DESCRIPTION
**Pull request by parruda**
## What does this PR do?
Makes secrets and context available within components via `event.secrets` and `event.context`.

## Do you have any concerns about this body of work? Anything you'd like to highlight or open for discussion? Anything to be aware of before merging / deploying?
nope

## Where should the review start?
`lib/gotta/serve/server.rb`

## Recommendation/Steps for how to test this
Create a project, edit the secrets file and add some stuff there for the root component.
Then change the root component code to `render json: event.secrets` and curl to see the secrets. Then to the same for `event.context`

## Link to the related issue(s)
Closes #9 
Closes #10

### CODE REVIEW

**Comment by tylerklose** [comment]
Edited the description so it will automatically close #10. I don't think `Closes #9 and #10` will work

**Comment by tylerklose** [comment]
@parruda what's the purpose of `context` and what kinds of things would I want to pass to my function via `context`? How would I set these variables?

File: lib/gotta/serve/server.rb [hunk 1]
```diff
@@ -45,7 +46,7 @@ class Server < Sinatra::Base
 
         query_params = parse_query(request.query_string)
 
-        secrets = {} # Gotta::Project.config.secrets[ENVIRONMENT]
+        secrets = Gotta::SecretsFile.load['secrets'][component_name]
```
Comment (tylerklose) [inline lib/gotta/serve/server.rb]: Have we discussed the idea of having a project-level secrets file component-level secrets file? The project-level secrets file would store any commonly used secrets while the component-level file would store secrets specific to the component and allow the ability to override the project-level secrets

Comment (parruda) [reply lib/gotta/serve/server.rb]: The secrets are currently component level.
We could add a special key `_global:` that would take global secrets.
There's also the alternative of using YAML syntax for that as well. (I lean more towards that one)

**Comment by parruda** [comment]
@tylerklose I basically renamed context to secrets, then reserved `event.context` for things we will provide, like environment name, maybe component name, things like that!

**Pull request merged**